instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb

```diff
@@ -150,7 +157,7 @@ def create_request_span_name(request_uri_or_path_info, env)
             end
           end
 
-          def set_attributes_after_request(span, status, headers, _response)
+          def set_attributes_after_request(span, status, headers, response)
```

Comment:
I'm concerned that we're now exposing the rack `response` body to the user, but I may be off base, I am not an expert on rack/rails tbh, but I recall that it's unreliable and potentially non performant to attempt to access the rack response body. For example with streaming response the behavior differs significantly. 

- https://tenderlovemaking.com/2011/03/03/rack-api-is-awkward.html 
- https://api.rubyonrails.org/classes/ActionController/Streaming.html
- https://github.com/rack/rack/blob/main/SPEC.rdoc#the-body-

Have you tested this against any practical examples or applications, for example? I'm not sure the rack `response` body makes sense to pass into the hook.